lib/verk/workers_manager.ex

```diff
@@ -193,6 +193,8 @@ defmodule Verk.WorkersManager do
         Log.start(job, worker)
         Verk.Worker.perform_async(worker, self, job)
         notify!(%Events.JobStarted{job: job, started_at: Timex.DateTime.now})
+      :full ->
+        Verk.enqueue(job)
```

Comment:
It's not that simple. We should never ask for jobs if we don't have a free worker. To push back a job we need to remove from the `inprogress` list as well. Dequeuing a job is done by this: https://github.com/edgurgel/verk/blob/master/priv/mrpop_lpush_src_dest.lua

So if we "need" to reenqueue a job I would rather consider this an invalid state that we got to and it's a bug.